pkg/controller/datavolume/import-controller.go

```diff
@@ -511,10 +319,14 @@ func (r *ImportReconciler) createVolumeImportSourceCR(syncState *dvSyncState) er
 	dv := syncState.dvMutated
 	importSource := &cdiv1.VolumeImportSource{}
 	importSourceName := volumeImportSourceName(dv)
+	isMultiStage := dv.Spec.Source != nil && len(dv.Spec.Checkpoints) > 0 &&
+		(dv.Spec.Source.VDDK != nil || dv.Spec.Source.Imageio != nil)
 
 	// check if import source already exists
-	if exists, err := cc.GetResource(context.TODO(), r.client, dv.Namespace, importSourceName, importSource); err != nil || exists {
+	if exists, err := cc.GetResource(context.TODO(), r.client, dv.Namespace, importSourceName, importSource); err != nil {
 		return err
+	} else if exists {
+		return r.updateVolumeImportSourceIfNeeded(importSource, dv, isMultiStage)
```

Comment:
Maybe rename `createVolumeImportSourceCR` to `reconcileVolumeImportSourceCR` since may create or update